src/command/subgraph/publish.rs

```diff
@@ -42,7 +52,49 @@ impl Publish {
         client_config: StudioClientConfig,
         git_context: GitContext,
     ) -> RoverResult<RoverOutput> {
+        // * if --allow-invalid-routing-url is not provided
+        // * AND a --routing-url is provided
+        // * AND the URL is unparsable
+        // we need to warn and prompt the user, else we can assume a publish
+        if !self.allow_invalid_routing_url {
+            if let Some(routing_url) = &self.routing_url {
+                if let Err(parse_error) = Url::parse(routing_url) {
+                    tracing::debug!("Parse error: {}", parse_error.to_string());
+
+                    if let Some(result) =
+                        Self::handle_invalid_routing_url(Self::prompt_for_publish(), None)
+                    {
+                        return Ok(result);
+                    }
+                }
+            }
+        }
+
+        // below is borrowed heavily from the `Fetch` command
         let client = client_config.get_authenticated_client(&self.profile)?;
+
+        if self.routing_url.is_none() {
+            let fetch_response = routing_url::run(
+                SubgraphRoutingUrlInput {
+                    graph_ref: self.graph.graph_ref.clone(),
+                    subgraph_name: self.subgraph.subgraph_name.clone(),
+                },
+                &client,
+            )?;
+
+            if let Some(routing_url) = fetch_response {
+                if let Err(parse_error) = Url::parse(&routing_url) {
```

Comment:
In the issue I laid out I said that we should just parse the URL and make sure it's valid, but I think we should probably do a few more things here to really make this experience shine:

1) check if the protocol in the URL is supported, and report that it should be `http` or `https`
2) check if the host is `localhost`, and if so, report that it is unroutable via the public internet